.github/workflows/tests-template.yml

```diff
@@ -0,0 +1,77 @@
+---
+name: Reusable Integration Workflow
+on:
+  workflow_call:
+    inputs:
+      processor:
+        required: true
+        type: string
+      gofailMake:
+        required: false
+        type: string
+        default: "no"
+      runs-on:
+        required: false
+        type: string
+        default: "['ubuntu-latest']"
+permissions: read-all
+
+jobs:
+  test-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - linux-${{ inputs.processor }}-unit-test-1-cpu
+          - linux-${{ inputs.processor }}-unit-test-2-cpu
+          - linux-${{ inputs.processor }}-unit-test-4-cpu
+          - linux-${{ inputs.processor }}-unit-test-4-cpu-race
+    runs-on: ${{ fromJson(inputs.runs-on) }}
```

Comment:
@ahrtr  it will be evaluated to `ubuntu-latest` for both `amd64` and `arm64` .. wdyt ?

Reply:
`runs-on` will need to be supplied as a parameter in the files that call this template. Refer https://github.com/etcd-io/etcd/blob/649b3659379c780ffa8f44723ac3410e1b922cfd/.github/workflows/robustness.yaml

Reply:
Consider using a smaller actuated runner size for bbolt though as I'm not sure these jobs need 8cpu and 32gb ram like etcd robustness does.

Reply:
+1, I just followed the same runners existed, it is just refactoring :) 

wdyt @ahrtr

Reply:
Let's start with 4cpu and 16gb, see how it's going.

Reply:
done